### DISCUSSION TITLE
Docker-compose and build-time data fetching

### DISCUSSION THREAD

**Discussion by john-prutton** [vercel/next.js#69055] [Help]
### Summary

I want to dockerize my whole app -- web app and database. I have a Dockerfile based on Vercel's example which works fine. The problem is composing the db and web app. When the web app builds, the db is not available/has not been started by docker. But Nextjs needs the db to be available during build so it can collect static page data.

Im considering spinning up the db separately using `docker-compose up postgres` and then spinning up the web app using `docker-compose up web`. But this just feels wrong.

Whats the best way to solve this?

### Proposed solutions

I found an article talking about this problem. Their solution was to move `npm build` from the Dockerfile into the started container using a entrypoint script.

https://dev.to/mattishida/a-gotcha-with-nextjs-production-builds-in-docker-compose-2232

**Comment by m-sanders** [comment]
There is an experimental split of the build and generate step. Take a look here:

https://github.com/vercel/next.js/discussions/46544

**Comment by Nithur-M** [comment]
Is there any solution to this yet?
